grammar.js

```diff
@@ -166,14 +167,27 @@ module.exports = grammar({
     ),
 
     function_statement: $ => seq(
+      optional($.procedure_attributes),
       optional($._callable_interface_qualifers),
       caseInsensitive('function'),
       $._name,
       optional($._parameters),
       optional($.function_result)
     ),
 
-    _callable_interface_qualifers: $ => repeat1(choice($.procedure_qualifier, $._intrinsic_type, $.derived_type)),
+    procedure_attributes: $ => seq(
```

Comment:
Can only a single procedure attribute be applied to a given function/subroutine? If more than one can be used we might want to move this rule into `_callable_interface_qualifers`.

Reply:
No, `attributes(host,device)` is also valid, but `attributes(host) attributes(device)` is not. Thanks for the suggestion, I went back and checked and for a function attributes can also be used after the type qualifier. Moved it to `_callable_interface_qualifiers`.